rest_framework_json_api/filters/filter.py

```diff
@@ -0,0 +1,135 @@
+import re
+import warnings
+
+from rest_framework.exceptions import ValidationError
+from rest_framework.settings import api_settings
+
+from rest_framework_json_api.utils import format_value
+
+# django-filter is an optional package. Generate a dummy class if it's missing.
+try:
+    from django_filters.rest_framework import DjangoFilterBackend
+except ImportError:
+    class JSONAPIDjangoFilter(object):
+
+        def __init__(self, *args, **kwargs):
+            """
+            complain that they need django-filter
+            TODO: should this be a warning or an exception?
+            """
+            warnings.warn("must install django-filter package to use JSONAPIDjangoFilter")
+
+        def filter_queryset(self, request, queryset, view):
+            """
+            do nothing
+            """
+            return queryset
+
+else:
+    class JSONAPIDjangoFilter(DjangoFilterBackend):
```

Comment:
I think we should follow the naming of upstream this way it is clear what we overwrite. In this case it would be `JSONAPIDjangoFilterBackend`.

Reply:
See my more general question about upstream naming in https://github.com/django-json-api/django-rest-framework-json-api/issues/467#issuecomment-416995999

Reply:
I guess pushing outdated this open review item. That was not my intention:-(

Reply:
Per https://github.com/django-json-api/django-rest-framework-json-api/issues/467#issuecomment-417338257 I've renamed to `DjangoFilterBackend` in 51b99464a7061f0c542c5eca42dfcf9fffc25467. This follows the upstream naming and is distinguished from it because it is imported from `rest_framework_json_api_filters`.

Reply:
I'm going to resolve this based on our discussion about naming in #471 
@sliverc let me know if you are OK and please merge.

Reply:
:+1: